src/librustc/ty/relate.rs

```diff
@@ -404,6 +404,19 @@ pub fn super_relate_tys<'a, 'gcx, 'tcx, R>(relation: &mut R,
             Ok(tcx.mk_generator(a_id, substs, interior))
         }
 
+        (&ty::TyGeneratorWitness(a_types), &ty::TyGeneratorWitness(b_types)) =>
+        {
+            // FIXME: Relate this without using tuples
```

Comment:
Can't you replicate the zip loop that the `TyTuple` branch does?

Reply:
I had to deal with the `Binder`, which needs an intermediate type. I could create one though.

Reply:
You could implement `Relate` for `&'tcx Slice<Ty<'tcx>>` using the same algorithm as `Substs`.

Reply:
I changed this to use an intermediate type to make it clear the logic only applies to TyGeneratorWitness.